### PR TITLE
tx simulation + config diff + minor refactor

### PR DESCRIPTION
**Pull request by sdrug**

### CODE REVIEW

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposeOffchainConfig.ts [hunk 1]
```diff
@@ -107,12 +118,33 @@ export default class ProposeOffchainConfig extends SolanaCommand {
...
+      offchainConfig,
+      process.env.SECRET!,
+      userSecret,
+    )
+    this.randomSecret = randomSecret
+
     return {
-      offchainConfig: ProposeOffchainConfig.makeInputFromRDD(rdd, this.args[0]),
+      offchainConfig,
+      serializedOffchainConfig,
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposeOffchainConfig.ts:145]: We can't ask the user to build the `serializedOffchainConfig`. This cannot be part of the `Input`.
An idea could be having `makeContractInput`, similar to what we have on Terra. The serialization would happen there.

**Review by RodrigoAD — COMMENTED** [review]
Looks good overall. Left some comments

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposal/acceptProposal.ts [hunk 2]
```diff
@@ -198,12 +198,15 @@ export default class AcceptProposal extends SolanaCommand {
...
+    const signer = this.wallet.publicKey
     const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
     const address = ocr2.programId.toString()
     const program = this.loadProgram(ocr2.idl, address)
 
-    const rawTx = await this.makeRawTransaction(this.wallet.publicKey)
+    const rawTx = await this.makeRawTransaction(signer)
+    await this.simulateTx(signer, rawTx)
     await prompt(`Continue accepting proposal of proposal ${this.flags.proposalId} on aggregator ${this.args[0]}?`)
+
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposal/acceptProposal.ts:209]: @sdrug We need a `beforeExecute` in this command to print the diff between the new proposal vs current config

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts [hunk 3]
```diff
@@ -24,17 +24,33 @@ export default class ProposePayees extends SolanaCommand {
...
+      (agg, operator) => ({
+        ...agg,
+        [new PublicKey(operator.transmitter).toString()]: new PublicKey(operator.payee),
+      }),
+      {},
+    )
+
+    const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
+    const address = ocr2.programId.toString()
+    this.program = this.loadProgram(ocr2.idl, address)
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts]: Can you move this into `buildCommand`?

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts [hunk 4]
```diff
@@ -80,35 +94,60 @@ export default class ProposePayees extends SolanaCommand {
...
+        ...agg,
+        [key]: {
+          transmitter: (value as any).transmitter,
+          payee: input?.payeeByTransmitter[(value as any).transmitter].toString(),
+        },
+      }),
+      {},
+    )
+
+    diff.printDiff(payeesInProposal, proposedPayees)
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts]: Let's provide some context to the user here printing some more message. Like `Proposing payees for contract ${this.args[0]}`.
This should be the message the user reads before executing.

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts [hunk 5]
```diff
@@ -80,35 +94,60 @@ export default class ProposePayees extends SolanaCommand {
...
+
   execute = async () => {
-    const rawTx = await this.makeRawTransaction(this.wallet.publicKey)
+    const signer = this.wallet.publicKey
+    const input = this.makeInput(this.flags.input)
+    await this.beforeExecute(input)
+
+    const rawTx = await this.makeRawTransaction(signer, input)
+    await this.simulateTx(signer, rawTx)
     await prompt('Continue setting payees proposal?')
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts]: Let's remove this and use the `beforeExecute` action only

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts (hunk 4 above)
Comment (RodrigoAD) [reply gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts]: Please check the rest of commands in this PR. Make sure all of them has at least a message indicating the action they are doing, and a prompt to ask for confirmation

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts (hunk 5 above)
Comment (RodrigoAD) [reply gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts]: Please check the rest of commands in this PR, make sure we use `beforeExecute` as user confirmation

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts [hunk 6]
```diff
@@ -76,39 +108,75 @@ export default class ProposePayees extends SolanaCommand {
...
+    const proposedPayees = Object.entries(payeesInContract).reduce(
+      (agg, [key, value]) => ({
+        ...agg,
+        [key]: {
+          transmitter: (value as any).transmitter,
+          payee: this.contractInput.payeeByTransmitter[(value as any).transmitter].toString(),
+        },
+      }),
+      {},
+    )
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts:164]: We have a small bug here, we are comparing the onchain vs onchain. I think should be:
```suggestion
    const proposedPayees = Object.entries(this.contractInput.payeeByTransmitter).reduce(
      (agg, [transmitter, payee], idx) => ({
        ...agg,
        [`operator#${idx}`]: {
          transmitter,
          payee: payee.toString(),
        },
      }),
      {},
    )
```

**Review by RodrigoAD — DISMISSED** [review]
I think is good to go. Please open the issue regarding the diff improvements

**Comment by sdrug** [comment]
Issue created here - https://github.com/smartcontractkit/chainlink-solana/issues/263
I still can't merge though, any ideas why those two checks are pending forever?

**Review by RodrigoAD — APPROVED** [review]

**Pull request merged**